R/conf_mat.R

```diff
@@ -142,28 +159,40 @@ conf_mat.grouped_df <- function(data,
     warn_conf_mat_dots_deprecated()
   }
 
-  names <- names(data)
+  truth <- enquo(truth)
+  estimate <- enquo(estimate)
+  case_weights <- enquo(case_weights)
 
-  truth <- tidyselect::vars_pull(names, {{truth}})
-  truth <- as.name(truth)
+  truth <- yardstick_eval_select(
+    expr = truth,
+    data = data,
+    arg = "truth",
+    error_call = error_call
+  )
+  estimate <- yardstick_eval_select(
+    expr = estimate,
+    data = data,
+    arg = "estimate",
+    error_call = error_call
+  )
 
-  estimate <- tidyselect::vars_pull(names, {{estimate}})
-  estimate <- as.name(estimate)
+  if (!quo_is_null(case_weights)) {
+    case_weights <- yardstick_eval_select(
+      expr = case_weights,
+      data = data,
+      arg = "case_weights",
+      error_call = error_call
+    )
 
-  case_weights <- enquo(case_weights)
-  if (quo_is_null(case_weights)) {
-    case_weights <- NULL
-  } else {
-    case_weights <- tidyselect::vars_pull(names, !!case_weights)
-    case_weights <- as.name(case_weights)
+    case_weights <- expr(.data[[!!case_weights]])
   }
 
   dplyr::summarise(
     data,
     conf_mat = {
-      table <- yardstick_table(
-        truth = !!truth,
-        estimate = !!estimate,
+      table <- conf_mat_impl(
+        truth = .data[[truth]],
+        estimate = .data[[estimate]],
         case_weights = !!case_weights
```

Comment:
```suggestion
        case_weights = .data[[!!case_weights]]
```

Would this have worked? Instead of using `expr()` above?

Reply:
Maybe you don't even need the `!!`?

Reply:
Couldn't find a good way to make it happen